### DISCUSSION TITLE
Penetration of the miner into rb scripts

### DISCUSSION THREAD

**Discussion by sandergol** [Homebrew/discussions#1214] [Everyday usage]
https://www.virustotal.com/gui/file/898eca6f30cc3ecaf2f96921407a181178c66ac4d0cbedc14b904eeafe95b4ed/detection

Not-a-virus:HEUR:RiskTool.Script.BitMiner.gen

Formula: orgalorg.rb
Path: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/orgalorg.rb

**Comment by SMillerDev** [comment]
I have no idea what you mean by this.
Is the upstream source compromised? -> report the issue to upstream
Is the ruby file a bitcoin miner? -> No it isn't, you can check the file on GitHub here: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/orgalorg.rb
Is it executing a bitcoin miner? -> report the issue upstream, we're only running `go build`

**Reply by gromgit** [reply]
From the Details tab in the URL that @sandergol posted:
```
Basic Properties
[...]
File type	JavaScript
Magic	ASCII C++ program text
```
Right...

**Comment by Bo98** [comment]
This is a false positive. I've managed to avoid it by doing this: https://github.com/Homebrew/homebrew-core/pull/74746

For some silly reason, some AVs see `-o 127.0.0.1` and erroneously cry "bit miner". This is a bug in their heuristics.

**Comment by sandergol** [comment]
Thank you for your submission.
The detection for this threat will be included in the next update of detection engine, expected version: 23100.

orgalorg.rb - Ruby/CoinMiner.A potentially unsafe application

Regards,

ESET Malware Response Team

**Reply by SMillerDev** [reply]
I'm not sure what you submitted, but as explained above: it's not a mining application. You're making antivirus checking worse for everyone.

**Reply by carlocab** [reply]
Looks like your report just broke `brew` for other users. https://github.com/Homebrew/homebrew-core/issues/74827

**Comment by JamesMatchett** [comment]
Hi all, 

Encountered this issue myself with ESET as a false positive as of today. Quick look on virus total showing that ESET is the only AV provider flagging this file as a threat (when looking at the source we can see that is not) 

https://www.virustotal.com/gui/file/3013cc52a0937d977faf53d53d86fc54d99954f5086a1c5e2cfa5bf3ffdce390/detection

Hopefully this will be rectified on ESET's side soon

**Reply by carlocab** [reply]
Yes, it seems @sandergol reported this to ESET as malware. Not really sure why they did that.

It might help if you reported this to ESET as a false positive.

**Comment by JamesMatchett** [comment]
Hi folks,

ticket was raised & I've gotten word it's been resolved by ESET as false positive, should not flag any more (hopefully)

**Reply by carlocab** [reply]
Thanks, @JamesMatchett.
